pkg/build/builder/cmd/builder.go

```diff
@@ -47,17 +48,32 @@ func newBuilderConfigFromEnvironment(out io.Writer) (*builderConfig, error) {
 
 	cfg.out = out
 
-	// build (BUILD)
 	buildStr := os.Getenv("BUILD")
-	glog.V(4).Infof("$BUILD env var is %s \n", buildStr)
 	cfg.build = &api.Build{}
-	if err := runtime.DecodeInto(kapi.Codecs.UniversalDecoder(), []byte(buildStr), cfg.build); err != nil {
-		return nil, fmt.Errorf("unable to parse build: %v", err)
+
+	obj, groupVersionKind, err := kapi.Codecs.UniversalDecoder().Decode([]byte(buildStr), nil, nil)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse build string: %v", err)
+	}
+	ok := false
+	cfg.build, ok = obj.(*api.Build)
+	if !ok {
+		return nil, fmt.Errorf("build string is not a build: %v", err)
+	}
+	if glog.V(4) {
+		redactedBuild := util.SafeForLoggingBuild(cfg.build)
+		if err != nil {
+			return nil, fmt.Errorf("unable to strip proxy credentials from build: %v", err)
+		}
+		bytes, err := runtime.Encode(kapi.Codecs.LegacyCodec(groupVersionKind.GroupVersion()), redactedBuild)
+		if err != nil {
+			return nil, fmt.Errorf("unable to serialize build: %v", err)
+		}
+		glog.V(4).Infof("redacted build: %v", string(bytes))
```

Comment:
yeah, the problem is printf %#v doesn't dereference nested pointers.  I also tried using spew instead, but it also didn't work for me so i punted and did this.  The nice part about doing it this way is the behavior should be exactly the same as it was before in terms of the format of the output.